modules/entity_version_workflows/src/EntityVersionWorkflowManager.php

```diff
@@ -71,19 +83,73 @@ public function updateEntityVersion(ContentEntityInterface $entity, $field_name)
     // from the transition.
     $current_state = $revision->moderation_state->value;
     $next_state = $entity->moderation_state->value;
-    /** @var \Drupal\workflows\TransitionInterface $transition */
-    $transition = $workflow_plugin->getTransitionFromStateToState($current_state, $next_state);
-    $values = $workflow->getThirdPartySetting('entity_version_workflows', $transition->id());
-    if (!$values) {
+
+    // Try to get the transition or do nothing.
+    try {
+      /** @var \Drupal\workflows\TransitionInterface $transition */
+      $transition = $workflow_plugin->getTransitionFromStateToState($current_state, $next_state);
+    }
+    catch (InvalidArgumentException $e) {
       return;
     }
 
+    $config_values = $workflow->getThirdPartySetting('entity_version_workflows', $transition->id());
+    if (!$config_values) {
+      return;
+    }
+
+    // If the config is set to "1" we check if the entity values has changed
+    // and then update the version numbers.
+    if (!empty($config_values['check_values_changed'])) {
+      if (!$this->isEntityChanged($entity)) {
+        return;
+      }
+      // Remove this to leave the version settings only for the iteration.
+      unset($config_values['check_values_changed']);
+    }
+
     // Execute all the configured actions on all the values of the field.
-    foreach ($values as $version => $action) {
+    foreach ($config_values as $version => $action) {
       foreach ($entity->get($field_name)->getValue() as $delta => $value) {
         $entity->get($field_name)->get($delta)->$action($version);
       }
     }
   }
 
+  /**
+   * Check if the entity has changed.
+   *
+   * @param \Drupal\Core\Entity\ContentEntityInterface $entity
+   *   The content entity object.
+   *
+   * @return bool
+   *   Return true if the entity has changed, otherwise return false.
+   */
+  protected function isEntityChanged(ContentEntityInterface $entity): bool {
+    if (empty($entity->original)) {
+      return FALSE;
+    }
+    $fields = array_keys($entity->toArray());
+    // Blacklist dynamic fields from the check.
+    $field_blacklist = [
+      'vid',
+      'revision_timestamp',
+      'moderation_state',
+    ];
+
+    // Let others change the blacklist.
+    $this->moduleHandler->alter('check_values_changed', $field_blacklist);
+
+    // Remove the blacklisted fields from checking.
+    $fields = array_diff($fields, $field_blacklist);
+    foreach ($fields as $field) {
+      // Check if the values are changed in the entity.
+      if ($entity->get($field)->hasAffectingChanges($entity->original->get($field)->filterEmptyItems(), $entity->language()->getId())) {
```

Comment:
This seems to be the cause for the following functional issue: after having a published revision if you create a new node and keep editing it, the minor version always bumps. This happen because the comparison is done with $entity->original which in this case doesn't seem to be target the last draft. On the fix we should introduce tests for this case.